activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -448,6 +448,7 @@ def find_with_ids(*ids)
         ids = ids.flatten.compact.uniq
 
         model_name = @klass.name
+        _select!(table[primary_key]) unless select_values.empty?
```

Comment:
Hum, actually, this mutates the current instance, I don't think we should do that. Can you do a quick followup that doesn't do this?

I think you can just do it in `find_some_ordered`, at the same point where `where(primary_key => ids)`.

Sorry, I should have been more careful.